Branch-SDK/Branch-SDK/BNCSpotlightService.m

```diff
@@ -0,0 +1,535 @@
+        activeController = ((UITabBarController *)rootViewController).selectedViewController;
+    } else {
+        activeController = rootViewController;
+    }
+    return activeController;
+}
+
+#pragma mark userActivity Delegate Methods
+
+- (void)userActivityWillSave:(NSUserActivity *)userActivity {
+    [userActivity addUserInfoEntriesFromDictionary:self.userInfo];
+}
+
+#pragma remove private content from spotlight indexed using Searchable Item
+
+- (void)removeSearchableItemsWithIdentifier:(NSString * _Nonnull)identifier
+                                   callback:(void (^_Nullable)(NSError * _Nullable error))completion {
+    if (identifier == nil) {
+        NSError *error = [NSError branchErrorWithCode:BNCSpotlightIdentifierError
+                                     localizedMessage:@"Spotlight indentifier not available"];
+        completion(error);
+        return;
+    }
+    [self removeSearchableItemsWithIdentifiers:@[identifier] callback:^(NSError * _Nullable error) {
+        if (completion)
+            completion(error);
+    }];
+}
+
+- (void)removeSearchableItemsWithIdentifiers:(NSArray<NSString *> *_Nonnull)identifiers
+                                    callback:(void (^_Nullable)(NSError * _Nullable error))completion {
+    
+    BOOL isIndexingAvailable = NO;
+    Class CSSearchableIndexClass = NSClassFromString(@"CSSearchableIndex");
+    SEL isIndexingAvailableSelector = NSSelectorFromString(@"isIndexingAvailable");
+    isIndexingAvailable =
+    ((BOOL (*)(id, SEL))[CSSearchableIndexClass methodForSelector:isIndexingAvailableSelector])
+    (CSSearchableIndexClass, isIndexingAvailableSelector);
+    
+    #define IndexingNotAvalable() { \
+        NSError *error = [NSError branchErrorWithCode:BNCSpotlightNotAvailableError];\
+        if (completion) {\
+            completion(error);\
+        }\
+        return;\
+    }
+    
+    if (!isIndexingAvailable) {
+        IndexingNotAvalable();
+    }
+    else {
+        if (CSSearchableIndexClass &&
+            [CSSearchableIndexClass respondsToSelector:@selector(defaultSearchableIndex)]){
+            id defaultSearchableIndex = [CSSearchableIndexClass defaultSearchableIndex];
+            if ([defaultSearchableIndex respondsToSelector:@selector(deleteSearchableItemsWithIdentifiers:completionHandler:)]) {
+                [defaultSearchableIndex deleteSearchableItemsWithIdentifiers:identifiers
+                                                           completionHandler:^(NSError * _Nullable error) {
+                                                               if (completion)
+                                                                   completion(error);
+                                                           }];
+            }
+            else {
+                IndexingNotAvalable();
+            }
+        }
+        else {
+            IndexingNotAvalable();
+        }
+    }
+    
+#undef IndexingNotAvalable
```

Comment:
It also seems inefficient to `#undef` this here and then `#define` it again in the next method. Since this is a .m, the macro is not globally visible. You could just define it at the top or anywhere above these two methods and then use it in both places.